docs/web-apps/automated-testing/pre-run-executables.md

```diff
@@ -0,0 +1,308 @@
+echo If objXMLHTTP.Status = 200 Then >> C:\dl.vbs
+echo Set objADOStream = CreateObject("ADODB.Stream") >> C:\dl.vbs
+echo objADOStream.Open >> C:\dl.vbs
+echo objADOStream.Type = 1 'adTypeBinary >> C:\dl.vbs
+echo objADOStream.Write objXMLHTTP.ResponseBody >> C:\dl.vbs
+echo objADOStream.Position = 0 'Set the stream position to the start >> C:\dl.vbs
+echo Set objFSO = Createobject("Scripting.FileSystemObject") >> C:\dl.vbs
+echo If objFSO.Fileexists(strHDLocation) Then objFSO.DeleteFile strHDLocation >> C:\dl.vbs
+echo Set objFSO = Nothing >> C:\dl.vbs
+echo objADOStream.SaveToFile strHDLocation >> C:\dl.vbs
+echo objADOStream.Close >> C:\dl.vbs
+echo Set objADOStream = Nothing >> C:\dl.vbs
+echo End if >> C:\dl.vbs
+echo Set objXMLHTTP = Nothing >> C:\dl.vbs
+
+cscript.exe C:\dl.vbs
+```
+
+</TabItem>
+
+<TabItem value="Linux">
+
+### Linux
+This shell script downloads **file.txt** at **mywebsite.com** to the **/tmp** directory.
+
+```js
+#!/bin/bash
+wget -O /tmp/file.txt http://mywebsite.com/file.txt
+```
+</TabItem>
+</Tabs>
+
+## Set Prerun Capability
+After you've created the download script, use the prerun capability in your test script to point to its location.
+
+If your script is in a publicly accessible location, you need to add the URL to the prerun capability.
+
+```js
+capabilities['prerun'] = "http://location.of/curl.sh"
+```
+
+## Editing the VM's Host File
+>**NOTE:** Editing the Host file of the virtual machine will not work if [Sauce Connect Proxy](https://wiki.saucelabs.com/display/DOCSDEV/Sauce+Connect+Proxy) is in use. If you are using Sauce Connect Proxy, the Host file of the machine running Sauce Connect Proxy will be referenced and you can make the desired changes there.
+
+An example of configuring a Sauce Labs virtual machine with a pre-run executable is editing the host file in the virtual machine, so when the driver tries to access a particular domain, like google.com, it will be redirected to a new IP address, for example 162.222.75.243 ([saucelabs.com](http://saucelabs.com/)). As with other `prerun` configurations, the basic steps are:
+
+1. Write a script with the URL redirect to the new IP address.
+1. Upload the script to a publicly accessible location, like GitHub or Sauce Storage
+1. Set the [`prerun` capability](https://wiki.saucelabs.com/display/DOCSDEV/Test+Configuration+Options#TestConfigurationOptions-Pre-runExecutables) in your test script to load the script as host file in the Sauce Labs virtual machine.
+
+### Host File Script
+Here are examples of the host file script, `EditDNS`, in both OS X/Linux and Windows versions.
+
+OS X/Linux Host File Script
+#!/bin/bash
+echo "162.222.75.243 www.google.com" >> /etc/hosts
+Windows Host File
+@echo off
+echo 162.222.75.243 www.google.com > %temp%\temphosts.txt
+type C:\WINDOWS\system32\drivers\etc\hosts >> %temp%\temphosts.txt
+copy /Y %temp%\temphosts.txt C:\WINDOWS\system32\drivers\etc\hosts
+
+## Setting the `prerun` Capability in Your Test Script
```

Comment:
Now I am actually super confused about this, since this section is here again, but as a H2 instead of an H3 inside the bigger Editing the VM's host file, although it actually does seem to go with that content, so maybe it is just a typo?

Anyway...I am still sticking with my suggestion that we only have one version of this section. Plus, this example, unlike the previous code sample, show waaaaaay more of the file than just the `prerun` setting, which it overkill, imo.